### PR TITLE
Take Flight!

### PR DESCRIPTION
**Pull request by paulswartz**
[Flight](https://flightjs.github.io) is a lightweight web framework that seems like it's a good fit for what Finda is trying to be.  I've spent a little time putting together a version which uses it to organize the various pieces.

I think it'll make it easier to develop and test different pieces going forward, and it also made it easy for me to support IE7 (which the current version didn't quite do).

You can see the results on my version: https://paulswartz.github.io/filter-template/

### CODE REVIEW

**Comment by ohnorobo** [comment]
I really like the separation of concerns here, but it's also introduced some bugs.

After fiddling abound for a little I'm seeing:
- deselected icons don't turn back to blue (clicking the icon again does changes it back)
- selecting an icon causes a 'jerky' zoom in if you're not already at the correct zoom level
- infobox never opens in chrome or firefox
- no searchbox in chrome or firefox

firefox:26.0
chrome:34.0.1818.0
safari:7.0 (9537.71)

I didn't look at IE.

**Comment by paulswartz** [comment]
I don't see the issues with the infobox/search box on Chrome or Firefox here. Are there errors in your console about anything?

**Comment by ohnorobo** [comment]
Uncaught ReferenceError: google is not defined search.js:10
I'm not seeing any console errors for the infobox.
That was looking at https://paulswartz.github.io/filter-template/

I pulled down your flight branch and the search/infobox look fine if I run locally, so it might just be something changing in the diff between your flight and gh-pages branches.

**Comment by paulswartz** [comment]
Ah, it's probably an issue with the Google geolocator not being loaded before the search box references it.  I'll add a test/fix for that as well.

**Comment by paulswartz** [comment]
@ohnorobo I think I've addressed those issues, but let me know if anything else comes up.

**Comment by ohnorobo** [comment]
I'm seeing some interference between geosearch and https on chrome and firefox

Chrome:
[blocked] The page at 'https://paulswartz.github.io/filter-template/' was loaded over HTTPS, but ran insecure content from 'http://maps.google.com/maps/api/js?sensor=false&callback=googleMapsApiLoaded&_=1391471338021': this content should also be loaded over HTTPS.

Firefox:
Blocked loading mixed active content "http://maps.google.com/maps/api/js?sensor=false&callback=googleMapsApiLoaded&_=1391471920247"

Everything works locally (on http). Not zooming on search is a nice change.

File: src/search.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          this.maxBounds = config.map.maxBounds;
+          this.addGoogleScript();
+        }
+      };
+
+      this.addGoogleScript = function() {
+        // separate function so we can mock it for testing
+        this.script = $('<script />')
+          .attr('type', 'text/javascript')
+            .attr('src', 'http://maps.google.com/maps/api/js?sensor=false&callback=googleMapsApiLoaded');
```
Comment (ohnorobo) [inline src/search.js]: Changing this to https should be enough.

**Comment by ohnorobo** [comment]
that works

**Comment by paulswartz** [comment]
I get other SSL errors on github.io, but at least this should work on other hosts.

**Pull request merged**